mne/decoding/tests/test_ssd.py

```diff
@@ -135,13 +135,28 @@ def test_ssd():
     ssd.fit(X)
     X_denoised = ssd.apply(X)
     assert_array_almost_equal(X_denoised, X)
+    # denoised by low-rank-factorization
+    ssd = SSD(info, filt_params_signal, filt_params_noise,
+              n_components=n_components, sort_by_spectral_ratio=True)
+    ssd.fit(X)
+    X_denoised = ssd.apply(X)
+    assert (np.linalg.matrix_rank(X_denoised) == n_components)
 
     # Power ratio ordering
-    spec_ratio, _ = ssd.get_spectral_ratio(ssd.transform(X))
+    ssd = SSD(info, filt_params_signal, filt_params_noise,
+              n_components=None, sort_by_spectral_ratio=False)
+    ssd.fit(X)
+    spec_ratio, sorter_spec = ssd.get_spectral_ratio(ssd.transform(X))
     # since we now that the number of true components is 5, the relative
     # difference should be low for the first 5 components and then increases
     index_diff = np.argmax(-np.diff(spec_ratio))
     assert index_diff == n_components_true - 1
+    # check sort_by_spectral_ratio
+    ssd = SSD(info, filt_params_signal, filt_params_noise,
+              n_components=None, sort_by_spectral_ratio=True)
+    ssd.fit(X)
+    sorter_spec2 = ssd.sorter_spec
+    assert all(sorter_spec == sorter_spec2)
```

Comment:
Is this the test we need?
Does it test that sorting from fit is applied during transform [open question]?
Perhaps we could test it more directly like running everything without sorting and then taking the sorter from a second fit and setting it to the first before transforming, then running a third fit with sorting activated and then comparing the results.
Any thoughts?

Reply:
mhmm that could be one option. But I also think that we can test actually the bug itself. Like splitting data in training and testing, evaluating some performance, and check there is no drop off in such performance.

Reply:
Hi @dengemann  ! check how I have made it now! it is implemented in `test_ssd:test_sorting`. Let me know what do you think now.